tests/test_strider.py

```diff
@@ -6,11 +6,16 @@
 
 import pytest
 
-from strider.fetcher import StriderWorker
+# Set redis host before importing
+os.environ["REDIS_HOST"] = "fakeredis"
```

Comment:
Is there a more elegant way to handle this switch, or test configuration in general? Is this sort of thing what `conftest.py` is for?

Reply:
FastAPI has a settings object: [https://fastapi.tiangolo.com/advanced/settings/](https://fastapi.tiangolo.com/advanced/settings/)

We can define default values in a `config.py` file. FastAPI reads environment variables at startup and will override the settings using those. In addition, settings are mutable so we can do `settings.redis_host = 'fakeredis'` in the test file. That should be cleaner than overriding environment variables.

Reply:
This sounds like a good thing. Does it solve the problem here where the configuration must be interleaved with the imports?

Reply:
Unfortunately it does not. We're creating the redis database connection at import time, so we still have to set the settings and then run the import. I looked at alternatives to this. The FastAPI recommended way is to use their dependency injection/management infrastructure. Reading here if you're interested:

- [Dependency Intro](https://fastapi.tiangolo.com/tutorial/dependencies/)
- [Dependencies in testing](https://fastapi.tiangolo.com/advanced/testing-dependencies/)
- [Database testing](https://fastapi.tiangolo.com/advanced/testing-database/)

I'm not a fan of this structure at all. The TL;DR is we would have to add a dependency parameter to every route, and then pass the database object to every function that needs it. Also, the dependency parameter then shows up in the generated documentation which seems like a very confusing thing to have in our API docs.

Reply:
Yeah, that doesn't sound fun, though I thought they were excluded from the API docs. Another option is to wrap the redis things in a generator of some sort so that the module can be imported before the classes are built. But that sounds kludgy.